assets/events.py

```diff
@@ -0,0 +1,203 @@
+        self._check_params('icon', merge)
+        self._check_params('message', merge)
+        self._check_params('severity', merge)
+        self._check_params('title', merge)
+        self._check_params('type', merge)
+        self._check_params('tags', merge)
+
+        self._send_events()
+
+        metadata = []
+
+        metadata_path = os.path.join(target_dir, 'metadata')
+        with open(metadata_path, 'w') as metadata_file:
+            json.dump(metadata, metadata_file)
+
+        return {
+            'version': {'timestamp': '0'},
+            'metadata': metadata,
+        }
+
+    def _merge_source_params(self, source, params):
+        merge = source.copy()
+        merge.update(params)
+
+        return merge
+
+    def _check_params(self, name, location):
+        if name not in location:
+            log.error("%s must exist in the configuration" % name)
+            exit(1)
+        else:
+            setattr(self, name, location.get(name))
+
+    def _send_events(self):
+        payload = {
+            'icon': self.icon,
+            'message': self.message,
+            'severity': self.severity,
+            'tags': self.tags,
+            'title': self.title,
+            'type': self.type
+        }
+        headers = {
+            'content-type': 'application/vnd.cycloid.io.v1+json',
+            'Authorization':'Bearer %s' % self.token
+        }
+
+        r = requests.post('%s/organizations/%s/events' % (self.api_url, self.organization), data=json.dumps(payload), headers=headers)
+        log.debug(r.text)
+
+
+    def _login(self):
+        # Login with informations
+        payload = {'email': self.api_login, 'password': self.api_password}
+        headers = {'content-type': 'application/vnd.cycloid.io.v1+json'}
+
+        try:
+            r = requests.post('%s/user/login' % (self.api_url), data=json.dumps(payload), headers=headers)
+            self.token = r.json().get('data').get('token')
+        except:
+            log.error("There is an error on login, please check your configuration")
+            exit(1)
+
+    def run(self):
+        """Parse input/arguments, perform requested command return output."""
+        # Extract informations from the json
+        source = self.data.get('source', {})
+        params = self.data.get('params', {})
+        version = self.data.get('version', {})
+
+
+        # Ensure we are receiving the required parameters on the configuration
+        self._check_params('api_url', source)
+        self._check_params('api_login', source)
+        self._check_params('api_password', source)
+        self._check_params('organization', source)
+
+        self._login()
+
+        # Define which operation to perform
+        if self.command_name == 'check':
+            response = self.check_cmd(source, version)
+        elif self.command_name == 'in':
+            response = self.in_cmd(self.command_argument[0], source, version, params)
+        else:
```

Comment:
Can only the command_name be 'check', 'in', 'out' ?
Asking otherwise we might add a case 'out', and otherwise output an error.

Reply:
yes, it's in the concourse resource RFC ;)